### PR TITLE
Solve #757 [DRAFT]

### PR DESCRIPTION
**Pull request by professorabhay**
Make the required changes as per need in #757

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -343,6 +343,30 @@ async fn main() -> anyhow::Result<()> {
         .map(|labelpair| parse_label(labelpair))
         .collect::<anyhow::Result<HashMap<String, String>, anyhow::Error>>()
         .context("failed to parse labels")?;
+    let mut labels = HashMap::from([
```
Comment (thomastaylor312) [inline src/main.rs:346]: So when you declare this variable, you are actually overriding the `labels` value declared on 339. The other thing is that these are the default labels so they should be the ones you start from and then `extend` them with the other user provided data. Generally the order of precedence in most projects is:

1. Command line flags
2. Env vars
3. Defaults

So what you'll probably want to do is start with:

```rust
let mut labels = HashMap::from([
    ("hostcore.arch".into(), ARCH.into()),
    ("hostcore.os".into(), OS.into()),
    ("hostcore.osfamily".into(), FAMILY.into()),
]);
```

Then `.extend` it with the stuff from 339-345

File: src/main.rs [hunk 2]
```diff
@@ -343,6 +343,30 @@ async fn main() -> anyhow::Result<()> {
...
+            ("hostcore.arch".into(), ARCH.into()),
+            ("hostcore.os".into(), OS.into()),
+            ("hostcore.osfamily".into(), FAMILY.into()),
+        ]);
+        labels.extend(config.labels.clone().into_iter());
```
Comment (thomastaylor312) [inline src/main.rs:351]: This line is no longer needed because it is coming from `args` above

**Review by thomastaylor312 — CHANGES_REQUESTED** [review]
Great start here! There are a few changes needed around ordering. Let me know if you need any more clarification on Rust things!

**Comment by professorabhay** [comment]
hey @thomastaylor312, Kindly take a look now !

File: crates/host/src/wasmbus/mod.rs [hunk 3]
```diff
@@ -1766,31 +1766,7 @@ impl Host {
         } else {
             Arc::new(KeyPair::new(KeyPairType::Server))
         };
-
-        let mut labels = HashMap::from([
```
Comment (thomastaylor312) [inline crates/host/src/wasmbus/mod.rs:1770]: You'll need to make sure you update places that were using `labels` to use `config.labels` instead in this function. You'll also need to remove some old imports you no longer need. If you run `cargo check --workspace`, you should see all of the compile errors and the warnings of imports you should remove

File: src/main.rs [hunk 4]
```diff
@@ -336,13 +336,37 @@ async fn main() -> anyhow::Result<()> {
         policy_changes_topic: args.policy_changes_topic,
         policy_timeout_ms: args.policy_timeout_ms,
     };
-    let labels = args
+    let mut labels = HashMap::from([
+        ("hostcore.arch".into(), ARCH.into()),
```
Comment (thomastaylor312) [inline src/main.rs:340]: You'll probably need to add this imports here. Once again, `cargo check` is your friend!

**Review by thomastaylor312 — CHANGES_REQUESTED** [review]
The code itself is looking correct, but you're missing some imports and updates to where the labels are being used!

**Comment by thomastaylor312** [comment]
Hey @professorabhay, just circling back to see if you needed any help

**Comment by thomastaylor312** [comment]
Since it has been a bit since we’ve heard anything, I’m going to go ahead and close for now. Few free to reopen if you have more time!

**Pull request closed without merging**